File: a004(propose)/C++/propose.cpp
#include <iostream>

int main(void) {
  int year;
  while (std::cin >> year) {
    if(year % 400 == 0 || (year % 4 == 0 && year % 100 != 0)) {
      std::cout << "閏年" << '\n';
    } else {
      std::cout << "平年" << '\n';
    }
  }
  return 0;
}

File: a004(propose)/Python/propose.py


while True:
    try:
        year = int(input())
        if(year % 400 == 0 or (year % 4 == 0 and year % 100 != 0)):
            print("閏年", end = "\n")
        else:
            print("平年", end = "\n")
    except:
        break
